skbio/stats/distance/_cutils.pyx

```diff
@@ -293,3 +293,119 @@ def permanova_f_stat_sW_cy(TReal[:, ::1] distance_matrix,
             s_W += local_s_W/group_sizes[group_idx]
 
     return s_W
+
+from libc.math cimport isnan, sqrt, acos, fabs
+
+cimport numpy as cnp
+
+ctypedef fused floating:
+    cnp.float32_t
+    cnp.float64_t
+
+ctypedef cnp.float32_t float32_t
+ctypedef cnp.float64_t float64_t
```

Comment:
hdmedians supports float32 and float64 (according to its [readme](https://github.com/daleroberts/hdmedians)). But in our case only float64 will be provided as input. Therefore we can remove float32 support.